src/index.ts

```diff
@@ -157,6 +157,21 @@ app.get('/profile/:user_id?', async (req: Request, res: Response) => {
     res.status(200).json(user);
 });
 
+// Post route to update profile information of user_id from users table.
+app.post('/profile/edit/:user_id', async (req: Request, res: Response) => {
+    const user = await db('users')
+        .update({
+            first_name  : req.params.first_name,
+            last_name   : req.params.last_name,
+            gender      : req.params.gender,
+            dob         : req.params.dob,
+            introduction: req.params.introduction,
+        })
+        .where({user_id: req.params.user_id} || req.auth.user_id)
```

Comment:
Since we no longer have a `user_id` in `req.params` object, This line should be `.where({user_id: req.auth.user_id})`